### PR TITLE
Rewrite complex flow

### PR DESCRIPTION
**Pull request by gciavarrini**
Rewrite Complex Flow example available in the bash [script](https://github.com/parodos-dev/parodos/blob/c7016f8a3a11bc9793e297661275f1353a01295e/workflow-examples/run_examples.sh#L200-L300) using workflow-service SDK

### CODE REVIEW

File: workflow-examples/src/test/java/com/redhat/parodos/examples/integration/utils/ExamplesUtils.java [hunk 1]
```diff
@@ -26,34 +27,97 @@
...
+			if (asyncResult.getError() != null) {
+				fail("An error occurred while executing getProjectAsync: " + asyncResult.getError());
+			}
+		}
+		finally {
+			lock.unlock();
+		}
+	}
+
+	public static WorkFlowStatusResponseDTO waitAsyncStatusResponse(WorkflowApi workflowApi, String workFlowExecutionId)
```
Comment (pkliczewski) [inline workflow-examples/src/test/java/com/redhat/parodos/examples/integration/utils/ExamplesUtils.java:83]: do we want to unify this method and `waitProjectStart`?

Comment (gciavarrini) [reply workflow-examples/src/test/java/com/redhat/parodos/examples/integration/utils/ExamplesUtils.java]: @pkliczewski done! Can you please take a look?

**Review by pkliczewski — COMMENTED** [review]
Looks great! one small question

File: workflow-examples/src/test/java/com/redhat/parodos/examples/integration/ComplexWorkFlow.java [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+		WorkFlowResponseDTO workFlowResponseDTO = workflowApi.execute(workFlowRequestDTO);
+		assertEquals(WorkStatusEnum.COMPLETED, workFlowResponseDTO.getWorkStatus());
+		log.info("workflow finished successfully with response: {}", workFlowResponseDTO);
+		if (workFlowResponseDTO.getWorkFlowOptions() == null
+				|| workFlowResponseDTO.getWorkStatus() != WorkStatusEnum.COMPLETED) {
+			fail("There is no valid INFRASTRUCTURE_OPTION");
+		}
+
+		// log.info("The Following newOption : {}",
+		// workFlowResponseDTO.getWorkFlowOptions().getNewOptions());
```
[the remark this answers is not in the document]
Comment (gciavarrini) [reply workflow-examples/src/test/java/com/redhat/parodos/examples/integration/ComplexWorkFlow.java]: Argh! Deleted!

**Review by pkliczewski — APPROVED** [review]

**Pull request merged**